Modules/System/AL Rest Client/src/ALHttpContent.Codeunit.al

```diff
@@ -0,0 +1,261 @@
+        case true of
+            Content.IsText:
+                begin
+                    ContentText := Content;
+                    if ContentType = '' then
+                        ContentType := MimeTypeTextPlainTxt;
+                    ALHttpContent.SetContent(ContentText, ContentType);
+                end;
+            Content.IsJsonObject:
+                begin
+                    JsonObject := Content;
+                    ALHttpContent.SetContent(JsonObject);
+                end;
+            Content.IsJsonArray:
+                begin
+                    JsonArray := Content;
+                    ALHttpContent.SetContent(JsonArray);
+                end;
+            Content.IsJsonToken:
+                begin
+                    JsonToken := Content;
+                    ALHttpContent.SetContent(JsonToken);
+                end;
+            Content.IsXmlDocument:
+                begin
+                    XmlDoc := Content;
+                    ALHttpContent.SetContent(XmlDoc);
+                end;
+            Content.IsCodeunit:
+                begin
+                    TempBlob := Content;
+                    if ContentType = '' then
+                        ContentType := MimeTypeApplicationOctetStreamTxt;
+                    ALHttpContent.SetContent(TempBlob, ContentType);
+                end;
+            Content.IsInStream:
+                begin
+                    InStream := Content;
+                    if ContentType = '' then
+                        ContentType := MimeTypeApplicationOctetStreamTxt;
+                    ALHttpContent.SetContent(InStream, ContentType);
+                end;
+            else
+                Error(ContentNotSupportedErr);
+        end;
+    end;
+
+    /// <summary>Initializes a new instance of the ALHttpContent object with the specified HttpContent object.</summary>
+    /// <param name="Content">The HttpContent object.</param>
+    /// <returns>The HttpContent object.</returns>
+    /// <remarks>The HttpContent must be properly prepared including the Content-Type header.</remarks>
+    procedure Create(Content: HttpContent) ALHttpContent: Codeunit "AL Http Content"
+    begin
+        ALHttpContent.SetHttpContent(Content);
+    end;
+    #endregion
+
+    #region Public Methods
+
+    /// <summary>Sets the Content-Type header of the HttpContent object.</summary>
+    /// <param name="ContentType">The value of the header to add.</param>
+    /// <remarks>If the header already exists, it will be overwritten.</remarks>
+    procedure SetContentTypeHeader(ContentType: Text)
+    begin
+        if ContentType = '' then
+            Error(ContentTypeEmptyErr);
+        SetHeader('Content-Type', ContentType);
+    end;
+
+    /// <summary>Sets the Content-Encoding header of the HttpContent object.</summary>
+    /// <param name="ContentEncoding">The value of the header to add.</param>
+    /// <remarks>If the header already exists, it will be overwritten.</remarks>
```

Comment:
I don't think that the remark statement is true, because it is possible to have multiple values for "Content-Encoding" and the code under this procedure does not overwrite the whole "Content-Encoding", but adds a new one at the end of the list.

Reply:
Nice catch! I've changed the comment.